applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.cpp

```diff
@@ -281,124 +282,51 @@ void AdjointFiniteDifferencingBaseElement::CalculateSensitivityMatrix(const Vari
 {
     KRATOS_TRY;
 
-    if ( mpPrimalElement->GetProperties().Has(rDesignVariable) )
-    {
-        // define working variables
-        Vector RHS_unperturbed;
-        Vector RHS_perturbed;
-
-        ProcessInfo copy_process_info = rCurrentProcessInfo;
-
-        // Get perturbation size
-        const double delta = this->GetPerturbationSize(rDesignVariable);
-
-        // Compute RHS before perturbion
-        mpPrimalElement->CalculateRightHandSide(RHS_unperturbed, copy_process_info);
-
-        if ( (rOutput.size1() != 1) || (rOutput.size2() != RHS_unperturbed.size() ) )
-            rOutput.resize(1, RHS_unperturbed.size());
-
-        // Save property pointer
-        Properties::Pointer p_global_properties = mpPrimalElement->pGetProperties();
-
-        // Create new property and assign it to the element
-        Properties::Pointer p_local_property(Kratos::make_shared<Properties>(Properties(*p_global_properties)));
-        mpPrimalElement->SetProperties(p_local_property);
-
-        // perturb the design variable
-        const double current_property_value = mpPrimalElement->GetProperties()[rDesignVariable];
-        p_local_property->SetValue(rDesignVariable, (current_property_value + delta));
-
-        // Compute RHS after perturbation
-        mpPrimalElement->CalculateRightHandSide(RHS_perturbed, copy_process_info);
-
-        // Compute derivative of RHS w.r.t. design variable with finite differences
-        for(IndexType i = 0; i < RHS_perturbed.size(); ++i)
-            rOutput(0, i) = (RHS_perturbed[i] - RHS_unperturbed[i]) / delta;
-
-        // Give element original properties back
-        mpPrimalElement->SetProperties(p_global_properties);
-
-        //call one last time to make sure everything is as it was before TODO improve this..
-        mpPrimalElement->CalculateRightHandSide(RHS_perturbed, copy_process_info);
-    }
-    else
-        if ( (rOutput.size1() != 0) || (rOutput.size2() != 0) )
-            rOutput.resize(0,0,false);
+    // Get perturbation size
+    const double delta = this->GetPerturbationSize(rDesignVariable);
+    ProcessInfo copy_process_info = rCurrentProcessInfo;
```

Comment:
what are you trying to do here?
it is called copy but you are assigning it

Reply:
I am aware that this is not nice but it needed because ```CalculateSensitivityMatrix``` receives the process info as const. I did a renaming: it is now simply called ```process_info```.

Reply:
As far as i know this is calling the copy constructor, because the `copy_process_info` was not initialized before.